File: medium/0075_sort_colors/solution.cpp
#include <vector>

class Solution {
public:
  void sortColors(std::vector<int> &nums) {
    std::vector<int> colorsCounts(3, 0);

    for (const auto &num : nums)
      ++colorsCounts[static_cast<size_t>(num)];

    size_t color{};
    for (size_t i{}; i < nums.size(); ++i) {
      while (colorsCounts[color] == 0)
        ++color;

      nums[i] = color;
      --colorsCounts[color];
    }
  }
};

File: medium/0075_sort_colors/solution.py
from typing import List


class Solution:
    def sortColors(self, nums: List[int]) -> None:
        colorsCounts = [0] * 3

        for num in nums:
            colorsCounts[num] += 1

        color = 0
        for i in range(len(nums)):
            while not colorsCounts[color]:
                color += 1

            nums[i] = color
            colorsCounts[color] -= 1
